File: leetcode/1.two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let cache = {}
    for (let i = 0; i < nums.length; i++) {
        let num = cache[target - nums[i]]
        if (num >= 0) {
            if (num > i) {
                return [i, num]
            } else {
                return [num, i]
            }
        }
        cache[nums[i]] = i
    }
    return []
};

File: Code/LeetCode-Python/1.two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}
        for i, num in enumerate(nums):
            temp = target - num
            if temp in hash:
                return [hash[temp], i]
            hash[num] = i
        return None